File: backend/app/schemas/billing.py
from datetime import datetime
from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel


class VendorEarningResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: UUID
    vendor_id: UUID
    report_id: UUID
    request_id: UUID | None = None
    lender_id: UUID
    amount: Decimal
    earning_type: str
    month: str


class LenderPayableResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: UUID
    lender_id: UUID
    report_id: UUID
    request_id: UUID | None = None
    amount: Decimal
    payable_type: str
    status: str
    month: str


class InvoiceResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: UUID
    invoice_type: str
    organization_id: UUID
    amount: Decimal
    status: str
    month: str
    generated_at: datetime | None = None
    invoice_number: str | None = None
    line_items_count: int = 0
    notes: str | None = None


class InvoiceWithOrgResponse(InvoiceResponse):
    org_name: str = ""


class InvoiceStatusUpdate(BaseModel):
    status: str  # PaymentStatus value: BILLED or PAID


class BulkStatusUpdate(BaseModel):
    invoice_ids: list[UUID]
    status: str  # PaymentStatus value


class GenerateInvoicesRequest(BaseModel):
    month: str  # Format: YYYY-MM


class BillingEntryResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: UUID
    report_id: UUID
    request_id: UUID | None = None
    amount: Decimal
    entry_type: str
    created_at: datetime


class BillingEntriesWithInvoice(BaseModel):
    entries: list[BillingEntryResponse]
    invoice_number: str | None = None
    invoice_status: str | None = None


class InvoiceDetailResponse(InvoiceWithOrgResponse):
    entries: list[BillingEntryResponse] = []

File: frontend/src/types/billing.ts
export interface InvoiceResponse {
  id: string;
  invoice_type: string;
  organization_id: string;
  amount: string;
  status: string;
  month: string;
  generated_at: string | null;
  invoice_number: string | null;
  line_items_count: number;
  notes: string | null;
  org_name: string;
}

export interface BillingEntry {
  id: string;
  report_id: string;
  request_id: string | null;
  amount: string;
  entry_type: string;
  created_at: string;
}

export interface InvoiceDetailResponse extends InvoiceResponse {
  entries: BillingEntry[];
}

export interface BillingEntriesWithInvoice {
  entries: BillingEntry[];
  invoice_number: string | null;
  invoice_status: string | null;
}

export interface GenerateInvoicesResponse {
  count: number;
  invoices: {
    id: string;
    invoice_number: string | null;
    invoice_type: string;
    amount: string;
    status: string;
  }[];
}

export interface BulkStatusResponse {
  updated: string[];
  skipped: string[];
}
